python/enthic/app.py

```diff
@@ -178,23 +179,60 @@ def pre_cast_integer(probe):
     return str(probe) if probe.__class__ is int else probe if probe.isnumeric() is True else None
 
 
-def result_array(probe, limit, offset=0):
+def get_corresponding_ape_codes(ape_code):
+    """
+    APE codes are not saved in the original format.
+    This function returns codes that are in the database
+    and corresponds to the given APE code or it's sub-categories
+    It returns None if the given APE code does not exist
+
+        :param ape_code : the list of APE for the filter, comma separated
+
+        :return: list of corresponding APE_CODE in database
+    """
+    result = list()
+    for one_code in ape_code.split(","):
+        for i in APE_CODE:
+            if re.match(one_code,APE_CODE[i][0]):
+                result.append(i)
+    if not result:
+        return None
+    return result
+
+def result_array(probe, limit, ape_code, offset=0):
     """
     List the result of the search in the database.
 
        :param probe: A string to match.
        :param limit: Integer, the limit of result to return.
+       :param ape_code: List of APE codes to match or None
        :param offset: Integer, offset of the select SQL request.
     """
-    with application.app_context():
-        companies = fetchall("""SELECT siren, denomination, ape, postal_code, town
-                        FROM identity WHERE siren = %s
-                        OR denomination LIKE %s
-                        OR MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE)
-                        LIMIT %s OFFSET %s;""", (pre_cast_integer(probe), "{0}%".format(probe),
-                                                 "{0}%".format(probe), limit, offset))
-    return tuple(CompanyIdentity(*company).__dict__ for company in companies)
+    sql_query_select_part = "SELECT siren, denomination, ape, postal_code, town FROM identity"
+    sql_query_count = "SELECT  COUNT(*) FROM identity"
 
+    sql_query_probe_condition = '1'
+    if probe:
+        sql_query_probe_condition = "denomination LIKE {} OR MATCH(denomination) AGAINST ({} IN NATURAL LANGUAGE MODE)".format(
+                                    "'{0}%'".format(probe),
+                                    "'{0}%'".format(probe))
+        #If probe is an interger, it might be a siren number, so we add this condition to the query
+        if pre_cast_integer(probe) :
```

Comment:
Unnecessary space after probe. Please use an IDE with automatic code reformatting such as Pycharm or use autopep8.

Reply:
I ran autopep8 locally, and added a commit with all the fix, but the one splitting line to long.